<!--xml-->
<launch>
  <node name="Motion_Controller_Node" pkg="icarus_rover_rc" type="Motion_Controller_Node">
    <param name="DEBUG_MODE" value="1"/>  
	<param name="LOGGING_ENABLED" value="0"/>
	<param name="Operation_Mode" value="LIVE"/>
	<param name="Northing_Origin_m" value="4510204.909545"/>
	<param name="Easting_Origin_m" value="282054.543418"/>
  </node>
  
  <node name="Mavlink_Node" pkg="icarus_rover_rc" type="Mavlink_Node.py">
  </node>
  
  <node name="Sonic_Controller_Node" pkg="icarus_rover_rc" type="Sonic_Controller_Node">
    <param name="Operation_Mode" value="LIVE"/>
	<param name="DEBUG_MODE" value="1"/>  
	<param name="LOGGING_ENABLED" value="0"/>
    <param name="sc_device" value="/dev/ttyUSB1" />
    <param name="baudrate" value="115200"/>
    <param name="rate" value="50"/>
    <param name="ping_sensor_count" value="5"/>
    <param name="angle_min_deg" value = "-90"/>
    <param name="angle_max_deg" value = "90"/>
    <param name="angle_increment_deg" value = "45"/>
	<param name="beam_width" value="30"/>
    <param name="range_min_m" value = "0.1524"/>
    <param name="range_max_m" value = "6.477"/>
	<param name="points_per_beam" value="30"/>
  </node>

<node pkg="gmapping" type="slam_gmapping" name="slam_gmapping" output="screen">
     <remap from="scan" to="Sonic_Controller_Node/ICARUS_Sonar_Scan"/>
      <param name="map_udpate_interval" value="5.0"/>
      <param name="maxUrange" value="16.0"/>
      <param name="sigma" value="0.05"/>
      <param name="kernelSize" value="1"/>
      <param name="lstep" value="0.05"/>
      <param name="astep" value="0.05"/>
      <param name="iterations" value="5"/>
      <param name="lsigma" value="0.075"/>
      <param name="ogain" value="3.0"/>
      <param name="lskip" value="0"/>
      <param name="srr" value="0.1"/>
      <param name="srt" value="0.2"/>
      <param name="str" value="0.1"/>
      <param name="stt" value="0.2"/>
      <param name="linearUpdate" value="1.0"/>
      <param name="angularUpdate" value="0.5"/>
      <param name="temporalUpdate" value="3.0"/>
      <param name="resampleThreshold" value="0.5"/>
      <param name="particles" value="30"/>
      <param name="xmin" value="-50.0"/>
      <param name="ymin" value="-50.0"/>
      <param name="xmax" value="50.0"/>
      <param name="ymax" value="50.0"/>
      <param name="delta" value="0.05"/>
      <param name="llsamplerange" value="0.01"/>
      <param name="llsamplestep" value="0.01"/>
      <param name="lasamplerange" value="0.005"/>
      <param name="lasamplestep" value="0.005"/>
    </node>
</launch>
